src/Build/Evaluation/Expander.cs

```diff
@@ -311,6 +311,8 @@ private void FlushFirstValueIfNeeded()
 
         private readonly IFileSystem _fileSystem;
 
+        private readonly LoggingContext _loggingContext;
```

Comment:
There are existing methods in the `Expander` class that take `LoggingContext` as a parameter. Makes me wonder if we can simplify them now that the context is added as an instance field. Related to that, can we require `LoggingContext` be passed to the constructor or are there scenarios where we would still want to create the `Expander` without a logging context?

Reply:
We surely want LoggingContext as much as possible - but it's well beyond the scope of this PR.

There are about gazillion of codepaths that can create Expander and that'd need to be updated (it's a little exaggeration - it's only in order of hundreds :-)), some are quite nested and unfortunately there are some that cannot provide the LoggingContext - e.g. ToolsetReader (as it expands path variables even before starting to evaluate particular project).

In any case I'm just working on experimental changeset that'd have this addressed

Reply:
Btw. just to brag about it again :-) Here is the experimental change (not yet finallized) illustrating the extend of propagating LoggingContext to Expander: https://github.com/dotnet/msbuild/pull/10009/commits/f97bb85f94704a4af68cede76b6efe67317723ca